infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -153,6 +153,19 @@
     // Connectivity of any responsive nodes not responding. Updated by FailureDetector.
     private Set<String> unresponsiveNodesPeerView = Collections.emptySet();
 
+    /**
+     * Fetches and creates a copy of the Management Layout from the local datastore.
+     *
+     * @return Copy of the management layout from the datastore.
+     */
+    private Layout getManagementLayout() {
+        Layout l = serverContext.getManagementLayout();
```

Comment:
You can remove the synchronized from ServerContext.getManagementLayout and probably create a method called CopyCurrentManagementLayout.